src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -147,14 +150,30 @@ public int getOrdinal() {
   // Current buffer size
   private volatile float bufferSize;
 
-  // Reference to the Streaming Ingest channel that owns this buffer
-  @VisibleForTesting final SnowflakeStreamingIngestChannelInternal<T> owningChannel;
-
   // Names of non-nullable columns
   private final Set<String> nonNullableFieldNames;
 
-  AbstractRowBuffer(SnowflakeStreamingIngestChannelInternal<T> channel) {
-    this.owningChannel = channel;
+  // buffer's channel fully qualified name with database, schema and table
+  final String channelFullyQualifiedName;
+
+  // metric callback to report size of inserted rows
+  private final Consumer<Float> rowSizeMetric;
+
+  // Allocator used to allocate the buffers
+  final BufferAllocator allocator;
+
+  // State of the owning channel
+  final ChannelState channelState;
+
+  // ON_ERROR option for this channel
+  final OpenChannelRequest.OnErrorOption onErrorOption;
+
+  AbstractRowBuffer(BufferInitialisationContext initialisationContext, ChannelState channelState) {
```

Comment:
What about combine the `BufferInitialisationContext` and `ChannelState`, and rename `ChannelState` to `ChannelProperties`? This way all we left are `ChannelProperties` and `ChannelFlushContext`, where `ChannelProperties` is used to store the properties associated with a channel and for passing whatever is needed for the buffer, and `ChannelFlushContext` is an immutable object that is used during flush.

Reply:
`BufferInitialisationContext` is not needed to be in the channel. It is used only in the buffer.
Eventually we should create the buffer outside of the channel to simplify testing. All stuff in the `BufferInitialisationContext` is not needed in the channel at all. If we keep them in `ChannelState/ChannelProperties `, the `ChannelState` will have more than one responsibility that is now to share channel/buffer state (See [Single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)). More over readers will think that everything in the `BufferInitialisationContext` is needed in the `ChannelProperties` in the channel and the buffer. This will complicate understanding and moving/refactoring things, like it did when we introduced Parquet, me being the reader :)

I can remove `BufferInitialisationContext` and just pass all its fields to the buffer constructor if the number of classes is the concern.